h5py/h5t.pyx

```diff
@@ -957,13 +960,22 @@ cdef class TypeFloatID(TypeAtomicID):
         # Handle non-standard exponent and mantissa sizes.
         for size, finfo in sorted(available_ftypes.items()):
             nmant = finfo.nmant
-            if nmant == 63 and finfo.nexp == 15:
+            maxexp = finfo.maxexp
+            minexp = finfo.minexp
+            # workaround for numpy's buggy finfo on float128 on ppc64 archs
+            if size == 16 and MACHINE.startswith('ppc64'):
```

Comment:
Just for my information - where did you get these values from?  I'm writing a patch for numpy at the moment.